### ISSUE TITLE
Windows: Slow service start (fluent-package)

### ISSUE THREAD

**Issue by daipom** [fluent/fluent-package-builder#618]
# Abst

* Windows service startup is noticeably slower for fluent-package
* This problem can cause timeout errors (EventID 7000 or EventID 7009)
* This problem occurs in environments with multiple CPUs
* This is the problem of [win32-service](https://github.com/chef/win32-service) with Ruby 3 series being slow
  * https://github.com/chef/win32-service/issues/84

# Details

Windows service startup is noticeably slower for fluent-package.
I'm talking about the time to reach the `Running` state. In other words, the time to finish [StartServiceCtrlDispatcher](https://learn.microsoft.com/en-us/windows/win32/api/winsvc/nf-winsvc-startservicectrldispatchera).
For td-agent, it takes about 5 seconds, but for fluent-package it can take more than 20 seconds.

Windows service needs to finish [StartServiceCtrlDispatcher](https://learn.microsoft.com/en-us/windows/win32/api/winsvc/nf-winsvc-startservicectrldispatchera) in 30 seconds by default.
So, this problem can cause timeout errors.

EventID 7000

```
The testservice service failed to start due to the following error: 
The service did not respond to the start or control request in a timely fashion.
```

EventID 7009

```
A timeout was reached (30000 milliseconds) while waiting for the testservice service to connect.
```

# Reproduce

If you try to start the service several times, it will often take 10 to 20 seconds.

I confirm this problem in the following:

* Windows 10 Home 64bit
  * 4 core CPU, 8 logical processors (Intel(R) Core(TM) i7-8565U CPU @ 1.80GHz 1.99GHz)
  * 16GB RAM
* AWS EC2 instance Windows Server 2019 Base
  * t2.medium
  * t2.large
  * c5.large

Interestingly, this problem does not occur in t2.small EC2 instance type.
I think it occurs in environments with multiple CPUs.

**Comment by daipom** [comment]
https://github.com/chef/win32-service/pull/85 will fix this issue.

**Comment by kenhys** [comment]
it should be fixed via #630
